### DISCUSSION TITLE
problem with sample/subsys/nvs with stm32wle5xx

### DISCUSSION THREAD

**Discussion by ghost** [zephyrproject-rtos/zephyr#70109] [Q&A]
Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? -> stm32sle5jc
 - What have you tried to diagnose or workaround this issue? -> yes. no problem for code
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit? -> no

**To Reproduce**
See error -> no error, simply the program stops at while(1), nothing is written to memory and no reboot

**Expected behavior**
what you describe in your example

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows) -> Mac os Venture
 - Toolchain (e.g Zephyr SDK, ...) -> zephyr 3.4.0
 - Commit SHA or Version used -> 

thank you

**Comment by Laczen** [comment]
@gwongwen, when it stops at `while(1)` it has already done quite some things. No error was shown, but it has printed some messages?

**Comment by ghost** [comment]
hello and thank you for taking the time to respond to my request.
here is the message present on the console:

*** Booting Zephyr OS build zephyr-v3.4.0 ***
[00:00:00.003,000] <dbg> fs_nvs: nvs_recover_last_ate: Recovering last ate from sector 0
[00:00:00.017,000] <inf> fs_nvs: 2 Sectors of 2048 bytes
[00:00:00.023,000] <inf> fs_nvs: alloc wra: 0, 7c0
[00:00:00.028,000] <inf> fs_nvs: data wra: 0, a8
Id: 1, Address: 192.168.1.1
Id: 2, Key: ff fe fd fc fb fa f9 f8 
Id: 3, Reboot_counter: 0
Id: 4, Data: DATA
Id: 5, Longarray: 0 1 2 3 4 5 6 7 8 9 a b c d e f 10 11 12 13 14 15 16 17 18 19 1a 1b 1c 1d 1e 1f 20 21 22 23 24 25 26 27 28 29 2a 2b 2c 2d 2e 2f 30 31 32 33 34 35 36 37 38 39 3a 3b 3c 3d 3e 3f 40 41 42 43 44 45 46 47 48 49 4a 4b 4c 4d 4e 4f 50 51 52 53 54 55 56 57 58 59 5a 5b 5c 5d 5e 5f 60 61 62 63 64 65 66 67 68 69 6a 6b 6c 6d 6e 6f 70 71 72 73 74 75 76 77 78 79 7a 7b 7c 7d 7e 7f 

then, nothing happens, the system is frozen and if I reboot, the same message appears. I don't see what I can do. I work with a stm32wle5xx (256kB of flash memory with a partition of 4kB at the end of the flash (@3c000 to @3ffff)

**Reply by Laczen** [reply]
Hi @gwongwen, there seems to be nothing wrong with the writing to flash (if you would erase the storage area you would see a slight difference). There might be a problem with the clock setup that makes it wait way longer than expected. I would try reducing the sleep time (or be patient 😀)... or even remove the sleep time completely to find out what the problem is.

**Comment by ghost** [comment]
Hello, I have just removed the sleep time and It works well.

problem solved. Thank you again for your help.
